hibernate-core/src/main/java/org/hibernate/query/sqm/mutation/internal/SqmMutationStrategyHelper.java

```diff
@@ -179,4 +186,83 @@ private static void cleanUpCollectionTable(
 			);
 		}
 	}
+
+	/**
+	 * Translates the original delete predicate to be used in the id subquery
+	 * forcing the use of the table alias qualifier
+	 */
+	public static Predicate getIdSubqueryPredicate(
+			Predicate predicate,
+			EntityMappingType entityDescriptor,
+			TableGroup tableGroup,
+			SharedSessionContractImplementor session) {
+		if ( predicate instanceof FilterPredicate || predicate instanceof SqlFragmentPredicate ) {
+			return getBaseRestrictions( entityDescriptor, tableGroup, session ).get( 0 );
+		}
+		else if ( predicate instanceof Junction ) {
+			final Junction original = (Junction) predicate;
+			if ( original.getPredicates().size() > 1 ) {
+				final Junction junction = new Junction(
+						original.getNature(),
+						original.getPredicates(),
+						original.getExpressionType()
+				);
+				final Predicate secondToLastPredicate = junction.getPredicates().get( junction.getPredicates().size() - 2 );
+				final Predicate lastPredicate = junction.getPredicates().get( junction.getPredicates().size() - 1 );
+				int filterPredicateIndex = -1;
+				int fragmentPredicateIndex = -1;
+				if ( lastPredicate instanceof Junction ) {
```

Comment:
Can you explain that a little? I'm not sure what this is for.

Reply:
Add a comment about the case maybe. Also, after copying the predicates list, you'll have to change the code below a bit by adding `predicate = junction;`

Reply:
It's needed when there is both an explicit where in the mutation query and e.g. both a filter and a discriminator or a filter and a sql-restriction. Added a comment to clarify this.